### DISCUSSION TITLE
New User Questions (API reference, datasource usage, etc.)

### DISCUSSION THREAD

**Discussion by sameleff** [ghostfolio/ghostfolio#2596] [Q&A] [answered]
New Ghostfolio user here (been wanting to transition to a self-hosted personal finance management tool for awhile now, but news of Mint's impending demise has pushed me to speed up that process). I was able to get Ghostfolio running in Docker per the instructions but I have a few questions.

1. Are there any reference documents for usage of the API other than what is contained in the official Ghostfolio readme for the import function? Typically there is an /api/v1/docs location where an API user can view the expected API call format, required/optional parameters, and even test API calls. Does something like this exist already (whether as an extension to the api itself, or hosted on a standalone webpage)?

2. I see in schema.prisma that there are several datasource options...
A. When importing activity (manually, using the csv/json GUI importer, or via the API) how do I instruct Ghostfolio to connect using one datasource vs. another? I see in the example csv file YAHOO & MANUAL are used...I tried changing MANUAL to GOOGLE_SHEETS but this did not work. I looked in Market Data under the Admin Control panel, but didn't see an option to change the datasource there either.
B.  From #2090 I see that there is a DATA_SOURCES environment variable that can be used to enable additional datasources. This variable is not documented in the .example.env file...are there other environment variables that I should be aware of?
C. From other issues / discussions, it's clear that some of the datasources (e.g. ALPHA_VANTAGE) require their own API keys or other authentication. I assume these keys can/should be placed in the same .env file...are there specific variable names I should use? How do I know which variables are required for each datasource?

3. Is there a way to import Asset Class & Asset Sub Class to activity data when using the csv/json GUI importer or the API?

4. What exactly is the functionality from adding the URL of each Platform? Is it just to grab the favicon, or is it relevant anywhere else as well?

5. There has been some discussion on using plaid to automatically connect to US banks and add transactions directly into Ghostfolio using its API. Was this implemented? If so, is there a guide for how to set this up in Ghostfolio?

6. In Admin Control > Market Data, what is the difference between Gather Recent Data, Gather All Data, and Gather Profile Data? 

I appreciate any and all help! Also (though I know this post is already pretty lengthy) - generally, as a new user, I think it would be very very helpful to have a Wiki to describe some of the functionality of Ghostfolio. Obviously some of my questions are specific to the self-hosted variant, so I completely understand and appreciate that there is likely to be less support vs. the premium cloud-based subscription...however, a lot them relate to a basic understanding of how to use Ghostfolio. I have to imagine there are lot of other new users that have the same questions as me! Plus, as I mentioned above, with the news that Mint is disappearing at the end of the year, I'm sure Ghostfolio will see a dramatic increase in new years over the next several months.

**Comment by dtslvr** [comment, marked as answer]
Hi @sameleff

Thank you for your interest in [Ghostfolio](https://ghostfol.io) and your questions. We appreciate your engagement with our project.

To make it easier for everyone to find answers and keep the discussions more organized, I would like to suggest that you split these questions into separate discussions, following a Q&A-style approach similar to _Stack Overflow_. If there are already discussions related to your questions, please consider using those.

This way, each question can receive the attention it deserves, and it will be easier for others to contribute and search for solutions.

Furthermore, it’s worth noting that everything documented in our [README.md](https://github.com/ghostfolio/ghostfolio/blob/main/README.md) is safe to use. However, everything else is considered experimental and you may use it at your own discretion and risk.

Thank you for your understanding!

**Reply by sameleff** [reply]
Thanks for the reply! Definitely hear ya on the organization front...I'll mark this post as closed and begin others for each specific topic.
